[NorbertLoh] iP

NorbertLoh opened the pull request:
# COW Chatbot 🐮 🐄 
> mooooo

COW bot helps you keep tracks of your tasks with a milky twist!

It is,
- text based
- easy to use
- and most importantly **_CUTE_**

How to download it
1. download the latest release [here](https://github.com/NorbertLoh/ip/releases/tag/v0.1)
2. and run the jar file!

Features:
- [x] Managing tasks
- [ ] Remainders
- [ ] Cute cows GUI

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public static void main(String[] args) {
    new Cow("data/cow.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

NorbertLoh committed:
Renamed chatbot and added greetings

NorbertLoh committed:
created echo and exit

NorbertLoh committed:
added a TodoList and the ability to add Todo items into the list and display them

NorbertLoh committed:
renamed todo to Task and added mark as done and unmark as done

NorbertLoh committed:
added javadocs for TodoList

NorbertLoh committed:
Created sub tasks that inherits Task and completed ToDos, Events and Deadlines

NorbertLoh committed:
Added TextUiTesting with Expected and Inputs file

NorbertLoh committed:
Added exceptions and handled errors

NorbertLoh committed:
Added deletion and use of collections-ArrayList

NorbertLoh committed:
trim() commands

NorbertLoh committed:
Add class to save and load the Todolist into a .txt file

NorbertLoh committed:
Merge branch 'branch-Level-7'

NorbertLoh committed:
Add parsing of date and time and due command

NorbertLoh committed:
Merge branch 'branch-Level-8'

NorbertLoh committed:
Add comments

NorbertLoh committed:
Refactor the code to extract out more closely related code as
classes

NorbertLoh committed:
Merge branch 'branch-A-MoreOOP'

NorbertLoh committed:
Move files all into packages

NorbertLoh committed:
Merge branch 'A-Packages'

NorbertLoh committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

NorbertLoh committed:
Add Gradle, Junit test for TodoCommand and Parser class

NorbertLoh committed:
Merge branch 'A-Gradle'

NorbertLoh committed:
Add command usage to help list

NorbertLoh committed:
Add Javadoc comments to most of the methods

NorbertLoh committed:
Add find command and print list methods

NorbertLoh committed:
Merge branch 'A-JavaDoc'

NorbertLoh committed:
Merge branch 'Level-9'

NorbertLoh committed:
Add checkstyle and correct style violations

NorbertLoh committed:
Add checkstyle and correct style violations

NorbertLoh committed:
Merge branch 'A-CheckStyle'

NorbertLoh committed:
Add GUI with cow

NorbertLoh committed:
Merge branch 'Level-10'

NorbertLoh committed:
Remove all uploaded image and change duke to cow

tingxuanp commented on src/main/java/cow/commands/DeleteCommand.java:
```diff
@@ -0,0 +1,36 @@
+
+/** Creates a delete command object. **/
+public class DeleteCommand extends Command {
+    public static final String COMMAND_WORD = "delete";
```
Great that you adhered to the Java coding standard of using screaming snake case for constants, and using a common prefix for associated constants

tingxuanp commented on src/main/java/cow/commands/FindCommand.java:
```diff
@@ -0,0 +1,37 @@
+    public static final String COMMAND_EXAMPLE = "todo book";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Finds all task with argument in the description.\n"
+            + "Example: " + COMMAND_EXAMPLE;
```
Great that you adhered to the coding standard of having line breaks before an operator

tingxuanp commented on src/main/java/cow/commands/EventCommand.java:
```diff
@@ -0,0 +1,49 @@
+    private final String description;
+
+    /**
+     * Creates an EventCommand instance.
+     * @param description of the event.
+     * @param from Start of Event.
+     * @param to End of event.
```
This comment applies for all your Javadoc comments. There should be an empty line between the description and parameter section.

tingxuanp commented on src/main/java/cow/parser/Parser.java:
```diff
@@ -0,0 +1,170 @@
+package cow.parser;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import cow.commands.ByeCommand;
+import cow.commands.Command;
+import cow.commands.DeadlineCommand;
+import cow.commands.DeleteCommand;
+import cow.commands.DueCommand;
+import cow.commands.EventCommand;
+import cow.commands.FindCommand;
+import cow.commands.HelpCommand;
+import cow.commands.IncorrectCommand;
+import cow.commands.ListCommand;
+import cow.commands.MarkCommand;
+import cow.commands.TodoCommand;
+import cow.commands.UnmarkCommand;
+import cow.exceptions.CowExceptions;
+import cow.message.Messages;
+import cow.tasks.Deadlines;
```
I like how the ordering of import statements is very consistent and neat.

tingxuanp commented on src/main/java/cow/tasks/Deadlines.java:
```diff
@@ -0,0 +1,75 @@
+
+    private String printDateTime() {
+        return this.by.format(DateTimeFormatter.ofPattern("MMM d yyyy HHmm"));
+    }
```
Just a minor nit but I think it would be better if the spacing between methods could be consistent (leave a 1 line break in between).

dez2003 commented on src/main/java/cow/commands/DeadlineCommand.java:
```diff
@@ -0,0 +1,49 @@
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a Deadline item to the list.\n"
+            + "Example: " + COMMAND_EXAMPLE;
+
+    private final LocalDateTime by;
```
Maybe instead of naming the variable "by", something like "dueDate" would be a clearer description of the variable's purpose

dez2003 commented on src/main/java/cow/parser/Parser.java:
```diff
@@ -0,0 +1,170 @@
+     * @param args provided beside command.
+     * @return a due command.
+     */
+    private static Command prepDue(String args) {
```
I think it's better to put the full word "prepare" in the method name

```suggestion
    private static Command prepareDue(String args) {
```

dez2003 commented on src/main/java/cow/parser/Parser.java:
```diff
@@ -0,0 +1,170 @@
+     * @return the parse int.
+     * @throws CowExceptions exceptions due to invalid index.
+     */
+    private static int getMarkUnmarkInt(String args) throws CowExceptions {
```
I'm not really sure if this method names clearly describes what it does. I think something like "getIndexFromCommand" might work better

dez2003 commented on src/main/java/cow/tasks/Deadlines.java:
```diff
@@ -0,0 +1,75 @@
+/**
+ * Deadlines Task class.
+ */
+public class Deadlines extends Task {
```
It's better if the class name is single-valued since it's a blueprint for a deadline task

```suggestion
public class Deadline extends Task {
```

dez2003 commented on src/main/java/cow/todolist/TodoList.java:
```diff
@@ -0,0 +1,137 @@
+     * @param substring substring to find.
+     * @return a TodoList.
+     */
+    public TodoList getFilteredByDesc(String substring) {
```
```suggestion
    public TodoList filterByDescOrder(String substring) {
```

dez2003 reviewed (commented):
Overall, good job with the code quality! Just some minor changes to fix

NorbertLoh committed:
Add assertions to methods

There are a few methods which takes in an index or a string. These
methods should not take in an empty string or an index below -1.

In order to prevent any index out of bound errors, an assertion is
place before the index or strings are used.

NorbertLoh committed:
Improve code quality

Many comments were lacking an extra line. Some Ui methods were using
string literals. Some methods were violating SLAP principle.

As a step towards improving code quality, string literals were
extracted out as constants into Message and methods were extracted
out to follow SLAP principle. An extra line is added to all
comments.

NorbertLoh committed:
Update data file

NorbertLoh committed:
Merge pull request #1 from NorbertLoh/branch-A-Assertions

Add assertions to methods

NorbertLoh committed:
Merge remote-tracking branch 'origin/master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/cow/commands/DeleteCommand.java

NorbertLoh committed:
Merge pull request #2 from NorbertLoh/branch-A-CodeQuality

Improve code quality

NorbertLoh committed:
Merge branch 'master' into branch-A-Streams

NorbertLoh committed:
Merge pull request #3 from NorbertLoh/branch-A-Streams

Merge branch 'master' into branch-A-Streams

NorbertLoh committed:
Revert "Update data file"

This reverts commit 3d0ae16d26e99549c8f5bc5ca6e7e273e2a327d5.

NorbertLoh committed:
Merge branch 'branch-A-Assertions'

NorbertLoh committed:
Add .yml file

NorbertLoh committed:
Remove I/O test from yml

NorbertLoh committed:
Fix style errors

NorbertLoh committed:
Remove I/O tests

NorbertLoh committed:
Add recurring deadline command

NorbertLoh committed:
Merge tag 'B-RecurringTasks'

NorbertLoh committed:
Update GUI and include css

NorbertLoh committed:
Merge branch 'A-BetterGui'

NorbertLoh committed:
Add personality to text outputs

NorbertLoh committed:
Merge branch 'A-Personality'

NorbertLoh committed:
Add exception handling for invalid paths

NorbertLoh committed:
Merge branch 'A-MoreErrorHandling'

NorbertLoh committed:
Add more JUnit testing

NorbertLoh committed:
Merge branch 'A-MoreTesting'

NorbertLoh committed:
Add AI.md to keep track of AI usage

NorbertLoh committed:
Merge branch 'A-AiAssisted'

NorbertLoh committed:
Fix bugs in todolist:delete and style checks

NorbertLoh committed:
Add README

NorbertLoh committed:
Merge branch 'A-UserGuide'

NorbertLoh committed:
Update README and fix bugs

NorbertLoh committed:
Update README

NorbertLoh committed:
Merge remote-tracking branch 'origin/master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/cow/commands/DeleteCommand.java

NorbertLoh committed:
Update javadocs and minor bug fixes

NorbertLoh committed:
Revert "Update javadocs and minor bug fixes"

This reverts commit 558f6eecfb1b8f2d7364067ac9077c72074ca84b.

NorbertLoh committed:
Update README
